### PR TITLE
Adding NERSC Cori build script.

### PR DESCRIPTION
**Pull request by mcbennet**
## Proposed changes

Add build script for NERSC Cori. This recipe was last verified for haswell nodes on April 19, 2021. 

## What type(s) of changes does this code introduce?

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

NERSC Cori

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- N/A. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
Cori recipe is in the manual. Prefer either script or manual. Not both.
My quick thoughts prefer build script and replace manual entry with a line pointing to the scrpts.
@prckent what is your preference?

File: config/build_nersc_cori.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#  3) darshan/3.2.1                                     8) ugni/6.0.14.0-7.0.1.1_7.54__ge78e5b0.ari         13) job/2.2.4-7.0.1.1_3.50__g36b56f4.ari             18) PrgEnv-intel/6.0.5                               23) cray-hdf5-parallel/1.10.5.2
+#  4) craype-network-aries                              9) pmi/5.0.14                                       14) dvs/2.12_2.2.167-7.0.1.1_17.6__ge473d3a2         19) craype-mic-knl                                   24) cmake/3.14.4
+#  5) intel/19.0.3.199                                 10) dmapp/7.1.1-7.0.1.1_4.64__g38cf134.ari           15) alps/6.6.58-7.0.1.1_6.22__g437d88db.ari          20) cray-mpich/7.7.10                                25) gcc/7.3.0
+
+# KNL CPU Complex
+mkdir build_nersc_cori_knl_cmplx
+cd build_nersc_cori_knl_cmplx
+cmake -DQMC_SYMLINK_TEST_FILES=0 -DCMAKE_SYSTEM_NAME=CrayLinuxEnvironment -DQMC_COMPLEX=1 ..
+nice make -j 8
+ls -l bin/qmcpack_complex
```
Comment (ye-luo) [inline config/build_nersc_cori.sh:67]: Not intended for this PR but I think we should move to proper install.

**Comment by jtkrogel** [comment]
(Chandler made this PR at my request, so I'm solidly voting for the script route)

**Comment by prckent** [comment]
I support merging this and updating the documentation rst (either now or in a later PR) to directly include the actual script file for perfect sync. Once we have a working example it will be straightforward to update the remaining documentation.

File: config/build_nersc_cori.sh (hunk 1 above)
Comment (prckent) [reply config/build_nersc_cori.sh]: Hence I should get back to #3183 ...

**Review by prckent — APPROVED** [review]
Marking approved to indicate I am OK with this being merged as is.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Pull request merged**